modules/swagger-jaxrs2/src/main/java/io/swagger/v3/jaxrs2/integration/OpenApiServlet.java

```diff
@@ -80,4 +107,43 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
         }
 
     }
+
+    public static MultivaluedHashMap<String, String> getQueryParams(String queryString) {
+        MultivaluedHashMap<String, String> queryParameters = new MultivaluedHashMap<>();
+
+        if (StringUtils.isEmpty(queryString)) {
+            return queryParameters;
+        }
+
+        String[] parameters = queryString.split("&");
+
+        for (String parameter : parameters) {
+            String[] keyValuePair = parameter.split("=");
+            queryParameters.put(keyValuePair[0], Collections.singletonList(keyValuePair[1]));
```

Comment:
Since using `MultivaluedHashMap`, we only need to put the flat value instead of a list with the value here:
```suggestion
            queryParameters.put(keyValuePair[0], keyValuePair[1]);
```

Reply:
The MultivaluedHashMap only accepts an List<String> as value, to flattening it won't work. Digging deeper into how this is handeled in BaseOpenApiResource is see that we really would want a proper request query handler with URI Decoder like the one found in org.glassfish.jersey.uri.UriComponent#decodeQuery() and used by UriInfo that BaseOpenApiResource pulls inn. This seem to require adding an dependency to jersey-common. Is that OK, or should be just try to handle it all here.

Reply:
> The MultivaluedHashMap only accepts an List as value, to flattening it won't work.

See [`MultivaluedHashMapl#add(K, V)`](https://docs.oracle.com/javaee/7/api/javax/ws/rs/core/AbstractMultivaluedMap.html#add-K-V-).

My suggested change was incorrect and was still using `put()` instead of `add()`. 😄

Reply:
> This seem to require adding an dependency to jersey-common. Is that OK, or should be just try to handle it all here.

Given that we want to support all valid JAX-RS implementations and not only Jersey, I don't think this is an option.

Reply:
Turns out the getParmeterMap() method from the request gives me an map with all keys and values, but the values are String[]. So I have rewritten the method, added URI Decoding support, moved it into it's own class and added some tests.